Mathlib/Algebra/BigOperators/Basic.lean

```diff
@@ -2319,6 +2356,30 @@ theorem prod_subtype_mul_prod_subtype {α β : Type*} [Fintype α] [CommMonoid 
 #align fintype.prod_subtype_mul_prod_subtype Fintype.prod_subtype_mul_prod_subtype
 #align fintype.sum_subtype_add_sum_subtype Fintype.sum_subtype_add_sum_subtype
 
+@[to_additive]
+lemma prod_ite_exists (p : ι → Prop) [DecidablePred p] (h : ∀ i j, p i → p j → i = j) (a : α) :
+    ∏ i, ite (p i) a 1 = ite (∃ i, p i) a 1 := by simp [prod_ite_one univ p (by simpa using h)]
+
+variable [DecidableEq ι]
+
+@[to_additive (attr := simp)] lemma prod_dite_eq (i : ι) (f : ∀ j, i = j → α) :
+    ∏ j, (if h : i = j then f j h else 1) = f i rfl := by simp
```

Comment:
This should probably be about `mulSingle`

Reply:
These match lemmas under the `Finset` namespace. But, well spotted, I had forgotten the `Pi.mulSingle` lemmas.

Reply:
That's prone to regolfing. I've added some "See also" instead (although they look a bit stupid since we're not going to put a doc
`Fintype.X` now has

Reply:
I think a single-lemma proof is more valuable than `by simp`, so I don't buy the regolfing argument. See also in the docstring isn't checked by CI and doesn't link in VSCode. It also means that if someone refactors the `Finset` lemmas, CI reminds them to adjust the `Fintype` ones too.

Reply:
What about that idea on Zulip to have a tag for lemmas which are not tagged simp because they are provable by simp?

Reply:
I have reworked the proofs the way you wanted